src/output_writers.jl

```diff
@@ -270,126 +270,215 @@ end
+    - `slice_kw`            : `dimname = Union{OrdinalRange, Integer}` will slice the dimension `dimname`.
+                              All other keywords are ignored.
+                              e.g. `xC = 3:10` will only produce output along the dimension `xC` between
+                              indices 3 and 10 for all fields with `xC` as one of their dimensions.
+                              `xC = 1` is treated like `xC = 1:1`.
+                              Multiple dimensions can be sliced in one call. Not providing slices writes
+                              output over the entire domain.
+"""
+
+function NetCDFOutputWriter(model, outputs; interval=nothing, frequency=nothing, filename=".",
+                        clobber=true, globalattrib=Dict(), outputattrib=nothing, compression=0, slice_kw...)
 
-    isfile(filepath) && rm(filepath)
+    validate_interval(interval, frequency)
 
-    nccreate(filepath, "u", "xF", xC, xC_attr,
-                            "yC", yC, yC_attr,
-                            "zC", zC, zC_attr,
-                            atts=u_attr, compress=fw.compression)
+    mode = clobber ? "c" : "a"
 
-    nccreate(filepath, "v", "xC", xC, xC_attr,
-                            "yF", yC, yC_attr,
-                            "zC", zC, zC_attr,
-                            atts=v_attr, compress=fw.compression)
+    # Initiates the output file with dimensions
+    write_grid(model; filename=filename, mode=mode,
+               compression=compression, attrib=globalattrib, slice_kw...)
 
-    nccreate(filepath, "w", "xC", xC, xC_attr,
-                            "yC", yC, yC_attr,
-                            "zF", zC, zC_attr,
-                            atts=w_attr, compress=fw.compression)
+    # Opens the same output file for writing fields from the user-supplied variable outputs
+    dataset = Dataset(filename, "a")
 
-    nccreate(filepath, "T", "xC", xC, xC_attr,
-                            "yC", yC, yC_attr,
-                            "zC", zC, zC_attr,
-                            atts=T_attr, compress=fw.compression)
+    # Creates an unliimited dimension "Time"
+    defDim(dataset, "Time", Inf); sync(dataset)
+    defVar(dataset, "Time", Float64, ("Time",)); sync(dataset)
+    len_time_dimension = 0 # Number of time-steps so far
 
-    nccreate(filepath, "S", "xC", xC, xC_attr,
-                            "yC", yC, yC_attr,
-                            "zC", zC, zC_attr,
-                            atts=S_attr, compress=fw.compression)
+    if isa(outputattrib, Nothing)
+        outputattrib = Dict("u" => Dict("longname" => "Velocity in the x-direction", "units" => "m/s"),
+                            "v" => Dict("longname" => "Velocity in the y-direction", "units" => "m/s"),
+                            "w" => Dict("longname" => "Velocity in the z-direction", "units" => "m/s"),
+                            "T" => Dict("longname" => "Temperature", "units" => "K"),
```

Comment:
Hmm, this will create difficulties for arbitrary tracers...

Reply:
Yeah, this should be addressed in another pull request where we create on object with the field and its associated attributes.